### PR TITLE
fix: Fail loud if messages can't be read for usage with `useExtracted`

### PR DESCRIPTION
**Pull request by amannn**
Prevent silent translation wipes by propagating read errors instead of returning empty arrays.

Previously, `loadLocaleMessages()` would silently catch all errors during file reads and return an empty array. This empty array would then be used by `saveLocale()` to overwrite all existing translations with empty strings. This change ensures that only `ENOENT` (file not found) errors result in an empty array (expected for new locales), while other read errors (e.g., file corruption, I/O issues) are now propagated, preventing accidental data loss.

---
<a href="https://cursor.com/background-agent?bcId=bc-5c8e526e-4861-411d-b241-a73664d14fa6"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-5c8e526e-4861-411d-b241-a73664d14fa6"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

---

> [!NOTE]
> Propagates non-ENOENT read/parse errors when loading message catalogs (only ENOENT returns empty), with tests and mocks updated to prevent silent translation wipes.
> 
> - **Extractor/Catalog loading**:
>   - `CatalogManager.loadLocaleMessages`: remove silent catch; propagate errors from persister.
>   - `CatalogPersister`:
>     - Return `[]` only on `ENOENT` in `read`; otherwise throw `Error while reading ...`.
>     - Throw `Error while parsing ...` on formatter parse failures.
>     - Add `getFileName` and use in path/error messages.
> - **Tests**:
>   - `ExtractionCompiler.test.tsx`:
>     - Add tests asserting propagation of read errors and parser errors; ensure only `ENOENT` yields empty.
>     - Enhance `fs/promises` mock to throw realistic `ENOENT` and support read interceptors.
>   - `MessageExtractor.test.tsx`: update inline snapshots (source map mapping and aliased `useTranslations`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit cd829fdbaa23966ea03db26796213cee364bfbd1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/next-intl/src/extractor/ExtractionCompiler.test.tsx [hunk 1]
```diff
@@ -2015,6 +2015,113 @@ msgstr "Hey!"
...
+    };
+
+    using compiler = createCompiler();
+
+    await expect(
+      compiler.compile(
+        '/project/src/Greeting.tsx',
+        filesystem.project.src['Greeting.tsx']
+      )
+    ).rejects.toThrow('Incomplete quoted string:\n> "Hal');
```
Comment (amannn) [inline packages/next-intl/src/extractor/ExtractionCompiler.test.tsx]: @cursoragent I'd like to include some meta information here like "Error while parsing en.po: Incomplete …".
